lib/chef/provider/user/mac.rb

```diff
@@ -143,27 +143,29 @@ def reload_user_plist
         #
 
         def create_user
-          cmd = [-"-addUser", new_resource.username]
-          cmd += ["-fullName", new_resource.comment] if prop_is_set?(:comment)
-          cmd += ["-UID", prop_is_set?(:uid) ? new_resource.uid : get_free_uid]
-          cmd += ["-shell", new_resource.shell]
-          cmd += ["-home", new_resource.home]
-          cmd += ["-admin"] if new_resource.admin
-
-          # We can technically create a new user without the admin credentials
-          # but without them the user cannot enable SecureToken, thus they cannot
-          # create other secure users or enable FileVault full disk encryption.
-          if prop_is_set?(:admin_username) && prop_is_set?(:admin_password)
-            cmd += ["-adminUser", new_resource.admin_username]
-            cmd += ["-adminPassword", new_resource.admin_password]
```

Comment:
it seems like a lot of functionality got deleted here as well

Reply:
Not really. In the original comment above, the concern was about secure token. On line 215-217, it checks the secure token and tries to toggle it:
```
          if diverged?(:secure_token)
            converge_by("alter SecureToken") { toggle_secure_token }
          end
```
And in `toggle_secure_token` implementation, it takes `admin_username` and `admin_password` and runs `sysadminctl` to enable / disable secure token for the user. As long as these as set, the secure token toggling should not be affected.

Another use case could be affected is that Chef does not run as `root` at all, so `sysadminctl` can use the provided admin user credential to create the user. I doubt anyone will run Chef as non-root user. But this will not work any way once switching to `dscl` (changes above), because `dscl` refuse to create users when running as non-root user. Error message like this,
```
<main> attribute status: eDSPermissionError
<dscl_cmd> DS Error: -14120 (eDSPermissionError)
```

Reply:
okay this is all starting to make more sense to me today on re-reading, and we can't really support non-root users well, and it seems particularly poor to try to have non-root users creating users as a use case.  doubt we'd ever manage to support that correctly.  it gets asked as a FAQ a lot, but but really running as root is a hard requirement.